sync/src/relayer/compact_block_process.rs

```diff
@@ -43,6 +44,20 @@ impl<'a, CS: ChainStore + 'static> CompactBlockProcess<'a, CS> {
         let compact_block: CompactBlock = (*self.message).try_into()?;
         let block_hash = compact_block.header.hash().to_owned();
 
+        let status = self.relayer.shared().get_block_status(&block_hash);
+        if status.contains(BlockStatus::BLOCK_STORED) {
+            return Ok(());
+        } else if status.contains(BlockStatus::BLOCK_INVALID) {
+            debug_target!(
+                crate::LOG_TARGET_RELAY,
+                "receive a compact block with invalid status, {:#x}, peer: {}",
+                block_hash,
+                self.peer
+            );
+            // TODO ban/punish this peer
```

Comment:
Punish should happen when insert `BLOCK_INVALID`.

Reply:
We should punish the peer in 2 scenarios:

  * when inserts `BLOCK_INVALID`
  * receive a block already marked as `BLOCK_INVALID` before

But I think "ban" is dangerous, we should do it in another PR, but not here. Maybe after refactoring the scoring mechanism. So I just leave a TODO.